### ISSUE TITLE
dart2js crash with --preview-dart-2 (-dev.60) - type 'JavaScriptSourceInformationStrategy<dynamic>' is not a subtype of type 'SourceInformationStrategy<Node>'

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/sdk#33328]
```
Dart2Js finished with:

web/main.dart:15:1:
Internal Error: The compiler crashed when compiling this element.
main() async {
^
The compiler is broken.

When compiling the above element, the compiler crashed. It is not
possible to tell if this is caused by a problem in your program or
not. Regardless, the compiler should not crash.

The Dart team would greatly appreciate if you would take a moment to
report this problem at http://dartbug.com/new.

Please include the following information:

* the name and version of your operating system,

* the Dart SDK build number (2.0.0-dev.60.0), and

* the entire message you see here (including the full stack trace
  below as well as the source location above).

The compiler crashed: type 'JavaScriptSourceInformationStrategy<dynamic>' is not a subtype of type 'SourceInformationStrategy<Node>'
#0      KernelSsaBuilder.build.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/kernel/kernel_backend_strategy.dart:109:29)
#1      CompilerTask.measure (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:63:44)
#2      KernelSsaBuilder.build (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/kernel/kernel_backend_strategy.dart:96:17)
#3      SsaBuilderTask.build (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/ssa/ssa.dart:92:21)
#4      SsaFunctionCompiler.compile (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/ssa/ssa.dart:42:29)
#5      JavaScriptBackend.codegen (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/js_backend/backend.dart:699:43)
#6      KernelCodegenWorkItem.run (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/kernel/kernel_backend_strategy.dart:80:21)
#7      Compiler.emptyQueue.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:467:70)
#8      CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178:35)
#9      Compiler.emptyQueue.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:467:32)
#10     CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178:35)
#11     Compiler.emptyQueue.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:465:28)
#12     CompilerDiagnosticReporter.withCurrentElement (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:793:15)
#13     Compiler.emptyQueue.<anonymous closure>.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:463:18)
#14     EnqueuerStrategy.processWorkItem (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/enqueue.dart:495:6)
#15     CodegenEnqueuer._forEach (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/js_backend/enqueuer.dart:241:20)
#16     CodegenEnqueuer.forEach (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/js_backend/enqueuer.dart:256:5)
#17     Compiler.emptyQueue.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:459:16)
#18     CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178:35)
#19     Compiler.emptyQueue (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:458:14)
#20     Compiler.processQueue.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:480:7)
#21     CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178:35)
#22     Compiler.processQueue (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:477:14)
#23     Compiler.compileLoadedLibraries.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:393:9)
#24     CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178:35)
#25     Compiler.compileLoadedLibraries (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:316:16)
#26     Compiler.runInternal (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:295:5)
<asynchronous suspension>
#27     Compiler.run.<anonymous closure>.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:221:38)
#28     new Future.sync (dart:async/future.dart:222:31)
#29     Compiler.run.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:221:20)
#30     CompilerTask.measureSubtask (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/common/tasks.dart:178:35)
#31     Compiler.run (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/compiler.dart:218:41)
#32     CompilerImpl.run.<anonymous closure>.<anonymous closure> (file:///b/build/slave/dart-sdk-mac-dev/build/sdk/pkg/compiler/lib/src/apiimpl.dart:148:22)
#33     _RootZone.runUnary (dart:async/zone.dart:1381:54)
#34     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#35     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:638:45)
#36     Future._propagateToListeners (dart:async/future_impl.dart:667:32)
#37     Future._completeWithValue (dart:async/future_impl.dart:482:5)
#38     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:512:7)
#39     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#40     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#41     _runPendingImmediateCallback (dart:isolate/runtime/libisolate_patch.dart:113:13)
#42     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:166:5)

```

**Comment by kevmoo** [comment]
Nope. Just need another build...

**Comment by kevmoo** [comment]
Fixed in c499e0021ad4050e17cb17ea3c22ab00eea5e92f
